internal/chezmoi/sourcestate.go

```diff
@@ -1403,6 +1403,9 @@ func (s *SourceState) addPatterns(
 			include = patternSetExclude
 		}
 		pattern := dir.JoinString(text).String()
+		if !strings.HasSuffix(pattern, "/**") {
+			pattern = pattern + "/**"
```

Comment:
If this fix is appropriate (no better solution exists) we might discuss where to put this pattern extension (`/**`).
It can be put either here, only applying to source state reads or directly into `patternset.go#Match` effecting all dependent "packages".

Reply:
I actually prefer to put it into `patternset.go#Match` as the behavior of `doublestar` is strange and unexpected
(`.config/i3` does not match `.config/i3/theme.conf`)  for ignoring files